src/components/Trade/hooks/useTradeRoutes/useTradeRoutes.ts

```diff
@@ -85,7 +85,10 @@ export const useTradeRoutes = (
 
   useEffect(() => {
     setDefaultAssets()
-  }, [assets, routeBuyAssetId, wallet, setDefaultAssets])
+    // We exclude setDefaultAssets from the dependency array here intentionally
+    // We dont want the default assets to be automaticaly set every time a new setDefaultAssets functon is instantiated
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [assets, routeBuyAssetId, wallet])
```

Comment:
Conceptually fine with this if it works for fixing what (?) this is supposed to fix, but I'm wondering in which case would we get a new `setDefaultAssets` reference ? `setDefaultAssets` should be referentially stable between renders.

If we're not talking renders, but multiple components consuming the `useTradeRoutes` hook, thus creating two different instances/contexts of it, then not/having this dependency would in theory not change anything, since two hooks instances won't share the same context

Reply:
looking into it a bit more